### PR TITLE
Table Styles

### PR DESCRIPTION
**Pull request by corykon**
Added some base css classes to use on `<table>` elements. Could use your eyes on this @andrew-frueh. 

For now, this is just styles, but I'd like to chat more with people about componentizing things at open space. Lots of people have built table components, and I think there are some good learnings to gather.

### CODE REVIEW

**Comment by andrew-frueh** [comment]
Wow.  For a visual perspective this is exceptional @corykon - very impressed.

How would your styles currently handle include a column of our hc-checkbox components (which would in turn highlight the row)?  Also how would your styles handle a column of hc-icon components (similar to [something like this](https://xd.adobe.com/view/f0c739fa-f83c-4cf5-ad04-d8f39a419fe7/) in Measures Manager)?

File: src/styles.scss [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+@import './app/lib/colors.scss';
+@import './app/lib/tables.scss';
+
 
 html,
 body {
     font-family: 'Open Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
-    margin: 0;    
+    margin: 0;
+    height: 100%;
```
Comment (benjanderson) [inline src/styles.scss]: this height 100% causes there to always be a scroll bar 

image

File: src/app/lib/tables.scss [hunk 2]
```diff
@@ -0,0 +1,144 @@
+@import "./colors.scss";
```
Comment (benjanderson) [inline src/app/lib/tables.scss]: I would recommend pulling some of the customizable parts into variables with !default so they can be customized if so desired https://robots.thoughtbot.com/sass-default.

File: src/app/lib/tables.scss [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+//-----------------------------------------
+
+table.hc-table {
+    position: relative;
+    width: 100%;
+    max-width: 100%;
+    color: $text;
+
+    td, th {
+        padding: 8px 16px;
```
Comment (benjanderson) [inline src/app/lib/tables.scss]: $cell-padding

File: src/app/lib/tables.scss [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+            &:nth-child(2n) {
+                background-color: $white;
+            }
+
+            &:nth-child(2n + 1) {
+                background-color: $slate-gray-100;
+            }
+
+            &:not(.empty-table-row):hover {
+                background-color: tint($primary-brand, 75%);
```
Comment (benjanderson) [inline src/app/lib/tables.scss]: $hover-color

File: src/app/lib/tables.scss [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        border-left: 1px solid $slate-gray-200;
+        border-right: 1px solid $slate-gray-200;
+
+        tr {
+            &:nth-child(2n) {
+                background-color: $white;
+            }
+
+            &:nth-child(2n + 1) {
+                background-color: $slate-gray-100;
```
Comment (benjanderson) [inline src/app/lib/tables.scss]: $row-color-alternate

File: src/app/lib/tables.scss [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    tbody {
+        font-size: 15px;
+        border-left: 1px solid $slate-gray-200;
+        border-right: 1px solid $slate-gray-200;
+
+        tr {
+            &:nth-child(2n) {
+                background-color: $white;
```
Comment (benjanderson) [inline src/app/lib/tables.scss]: $row-color

File: src/app/lib/tables.scss [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    // Sorting
+    th.hc-col-sortable, th.hc-col-sortable-left {
+        position: relative;
+        cursor: pointer;
+        padding-right: 35px;
+
+        &:hover {
+            background-color: tint($primary-brand, 75%);
```
Comment (benjanderson) [inline src/app/lib/tables.scss]: reuse

File: src/app/lib/tables.scss (hunk 2 above)
Comment (benjanderson) [reply src/app/lib/tables.scss]: also aliasing the font-awesome characters with variables would make it more readable

File: src/app/lib/tables.scss [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+    td, th {
+        &.hc-cell-borders {
+            border: 1px solid $slate-gray-200;
+        }
+    }
+
+    // Condensed
+    &.hc-table-small {
+        td, th { padding: 5px 16px; font-size: 13px;}
```
Comment (benjanderson) [inline src/app/lib/tables.scss]: $cell-padding-condensed

**Comment by corykon** [comment]
> How would your styles currently handle include a column of our hc-checkbox components (which would in turn highlight the row)? Also how would your styles handle a column of hc-icon components (similar to something like this in Measures Manager)?

Shouldn't have any problems implementing either of those cases on top of these styles!

File: src/app/lib/tables.scss (hunk 2 above)
Comment (corykon) [reply src/app/lib/tables.scss]: Great suggestions, I'll work on that

**Pull request merged**